core/src/main/java/org/veriblock/sdk/sqlite/tables/GenericBlockRepository.java

```diff
@@ -87,6 +87,8 @@ public void saveAll(List<Block> blocks) throws SQLException {
                 stmt.addBatch();
             }
             stmt.executeBatch();
+        } finally {
```

Comment:
Looks a bit like a hack to me.
I would prefer to switch to autoCommit = false and explicitly commit whenever we like.
At least save the current state and restore it instead of always set to autoCommit = true

Reply:
autoCommit = true is a default flag for our functionality.
I set autoCommit = false only inside this method (line:83) to implement Batch functionality. But I forgot set it back.

Reply:
Yes. But it may happen that someone decides to set connection to autoCommit = false. This line of code would revert it and it will take some time to find this bug.

Reply:
Right now we have opposite situation. All our methods used auto commit, except one. From my opinion handling commits by hand  will call more bugs then use auto commit.

Reply:
I mean that you can store the autoCommit state, set it to false and revert back after you do not need it anymore.

Reply:
okay